pkg/daemons/executor/embed_windows.go

```diff
@@ -0,0 +1,89 @@
+//go:build windows && !no_embedded_executor
+// +build windows,!no_embedded_executor
+
+package executor
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/sirupsen/logrus"
+
+	// registering k3s cloud provider
+	_ "github.com/k3s-io/k3s/pkg/cloudprovider"
+	daemonconfig "github.com/k3s-io/k3s/pkg/daemons/config"
+)
+
+const (
+	networkName = "flannel.4096"
+)
+
+type SourceVipResponse struct {
+	IP4 struct {
+		IP string `json:"ip"`
+	} `json:"ip4"`
+}
+
+func platformKubeProxyArgs(nodeConfig *daemonconfig.Node) map[string]string {
+	argsMap := map[string]string{}
+	argsMap["network-name"] = networkName
+	if sourceVip := waitForSourceVip(networkName, nodeConfig); sourceVip != "" {
+		argsMap["source-vip"] = sourceVip
+	}
+	return argsMap
+}
+
+func waitForSourceVip(networkName string, nodeConfig *daemonconfig.Node) string {
+	for range time.Tick(time.Second * 5) {
+		network, err := hcsshim.GetHNSNetworkByName(networkName)
+		if err != nil {
+			logrus.WithError(err).Warningf("can't find HNS network, retrying %s", networkName)
+			continue
+		}
+		if network.ManagementIP == "" {
+			continue
```

Comment:
Why would this happen? Because we have just created the network and it is not ready yet?

Reply:
yes, it could be a short period of time the IP is not available. I will add some logging to be explicit about this retry.